tests/scripts/all.sh

```diff
@@ -3479,6 +3479,10 @@ component_build_armcc () {
     scripts/config.py baremetal
     # armc[56] don't support SHA-512 intrinsics
     scripts/config.py unset MBEDTLS_SHA512_USE_A64_CRYPTO_IF_PRESENT
+
+    # stop armclang warning about feature detection for A64_CRYPTO
+    scripts/config.py unset MBEDTLS_SHA256_USE_A64_CRYPTO_IF_PRESENT
```

Comment:
Isn't this a regression? As far as I can tell, the feature detection and the accelerated sha256 code does work with Armclang. The warning is because we only compile for a bare-metal software environment, so there's no available feature detection. With `MBEDTLS_SHA256_USE_A64_CRYPTO_ONLY`, I checked locally that the build emits no warning.

Given that we do all armcc and armclang builds with the same configuration, I think it's ok to not try compiling for A64 code with armclang. But that's a limitation that we should at the very least document in the script, and preferably track. Filed as https://github.com/Mbed-TLS/mbedtls/issues/7198.

Please add a comment explaining why we're disabling the option or referencing the issue.

Reply:
I don't think this is a library regression. My understanding is that the behaviour is correct - there is no feature detection, and the warning lets the user know this. But for the purposes of `all.sh`, we don't consider this warning to be a problem with the build.

It's a testing regression in that we no longer build a configuration that builds correctly, but with a warning.

So the comment `stop armclang warning about feature detection` is accurate, I think. I will add a fuller explanation.